packages/components/src/border-control/test/index.js

```diff
@@ -22,28 +22,37 @@ const defaultBorder = {
 	width: '1px',
 };
 
-const props = {
-	colors,
-	label: 'Border',
-	onChange: jest.fn().mockImplementation( ( newValue ) => {
-		props.value = newValue;
-	} ),
-	value: defaultBorder,
-};
+function createProps( customProps ) {
+	const props = {
+		colors,
+		label: 'Border',
+		onChange: jest.fn().mockImplementation( ( newValue ) => {
+			props.value = newValue;
+		} ),
+		value: defaultBorder,
+		...customProps,
+	};
+	return props;
+}
 
 const toggleLabelRegex = /Border color( and style)* picker/;
 
-const renderBorderControl = ( customProps ) => {
-	return render( <BorderControl { ...{ ...props, ...customProps } } /> );
+const renderBorderControl = async ( props ) => {
```

Comment:
OK I added explanatory comment so that any reader will understand connection between the mysterious `await act` and `floating-ui`'s `useFloating` hook.

I don't want to spend time on inlining the `renderBorderControl` calls now. Some wrappers around the plain `render` and `rerender` calls still makes sense, because there is the additional `act()` call. We could refactor them to:
```js
renderAndWait( <BorderControl { ...props } /> );
```
making the wrapper a bit shorter, but let's do that in another PR. We could probably avoid all the `rerender` calls, too, by putting the `value` prop into local `useState` of a wrapper component that is actually tested. But let's do that independently, as it's not necessary for React 18 upgrade.

> inline comments would definitely help, because `act()` around `element.focus()` seems pretty redundant.

`element.focus()` is an issue unrelated to `floating-ui` and I'd say the need to wrap it in `act()` is not a mystery at all: we're directly calling a method of the DOM element object, which will internaly fire a `focus` event, completely circumventing any Testing Library wrappers like `fireEvent` or `userEvent`. I did a similar fix for another component in #45243.